daemon/cmd/hubble.go

```diff
@@ -191,10 +191,11 @@ func (d *Daemon) launchHubble() {
 		return
 	}
 	logger.WithField("address", sockPath).Info("Starting local Hubble server")
-	if err := localSrv.Serve(); err != nil {
-		logger.WithError(err).Error("Failed to start local Hubble server")
-		return
```

Comment:
There was no code relying on localSrv after this, so my thinking was by not returning, we allow more of the hubble server to start, minimizing how degraded the functionality is when it can't start 1 of the servers. 

Regarding the error, I actually kind of think Serve should probably just return the error and let the caller handle logging it. 

Additionally, I don't agree with the idea of having it call Serve() in s go routine. This is what it used to do, but Go conventions are that blocking calls should be launched in Go routines by the caller. This is actually why I moved much of the logic from Serve() into NewServer().

Reply:
> Additionally, I don't agree with the idea of having it call Serve() in s go routine. This is what it used to do, but Go conventions are that blocking calls should be launched in Go routines by the caller. This is actually why I moved much of the logic from Serve() into NewServer().

Not sure what you mean by this. I agree with your appraoch which moves the go routine to the caller, I think it's an improvement. My concern is solely about the inconsistent error handling and unused log lines.

Reply:
Ok, then we're on the same page. I'm going to refactor this again slightly to just return the error and then this code won't be unused and we'll have the logging done by the caller.